motionplan/constraint.go

```diff
@@ -493,3 +480,53 @@ func NewOctreeCollisionConstraint(octree *pointcloud.BasicOctree, threshold int,
 	}
 	return constraint
 }
+
+// NewBoundingRegionConstraint will determine if the given list of robot geometries are in collision with the
+// given list of bounding regions.
+func NewBoundingRegionConstraint(robotGeoms, boundingRegions []spatial.Geometry, collisionBufferMM float64) StateConstraint {
+	return func(state *ik.State) bool {
+		internalGeoms, err := obstacleAndBoudingRegionsConstraintHandler(robotGeoms, state, defaultBoundingRegionConstraintDesc)
+		if err != nil {
+			return false
+		}
+		cg, err := newCollisionGraph(internalGeoms, boundingRegions, nil, true, collisionBufferMM)
+		if err != nil {
+			return false
+		}
+		return len(cg.collisions(collisionBufferMM)) != 0
+	}
+}
+
+func obstacleAndBoudingRegionsConstraintHandler(
```

Comment:
I don't understand what this function is supposed to be doing.  It seems to be returning geometries but I wouldnt expect that based on the function name.  Furthermore its switching off of the passed in state and also the constraintType which is confusing.  

I think we're trying to do too much by making this all into one function.  If something doesnt neatly fit together I'd rather maintain two code paths to support it rather than trying to fit it into a box that doesnt make sense